### PR TITLE
96 plotly medication

### PR DESCRIPTION
**Pull request by SenseCode**
close issue#96,change medication gantt chart from zingchart  to plotly.

### CODE REVIEW

File: src/encoded/schemas/surgery.json [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+                  "enum": [
+                      "Laparoscopic",
+                      "Open"
+                  ]
+              },
+              "robotic_assist": {
+                  "title": "Robotic Assit",
+                  "type": "boolean"
+              },
+              "location": {
```
Comment (vsmalladi) [inline src/encoded/schemas/surgery.json]: @SenseCode  Move Location to same level as Surgery Type. Add "Not Available" option for Surgery location.

File: src/encoded/tests/data/inserts/surgery.json [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+      "robotic_assist": true,
+      "location": "UTSW"
+    },
+    "uuid": "717612f1-0fb7-411b-a8b3-b045e252d098"
+  },
+  {
+    "patient": "KCEPT708IJT",
+    "date": "1860-04-03",
+    "surgery_type": "Metastectomy",
+    "nephrectomy_details": {
```
Comment (vsmalladi) [inline src/encoded/tests/data/inserts/surgery.json]: Remove nephrectomy_details for Metastectomy and Ablation
surgery_type
location
date

File: src/encoded/schemas/surgery.json [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                  ]
+              }
+            }
+        }
+    },
+
+    "allOf": [
+        {
+            "properties": {
+                "surgery_type": {
```
Comment (vsmalladi) [inline src/encoded/schemas/surgery.json]: If surgery_type is not "Nephrectomy" lets not submit nephrectomy_details.

File: src/encoded/schemas/surgery.json [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                    "format": "date"
+                }
+            ]
+        },
+        "surgery_type": {
+            "title": "Surgery procedure",
+            "description": "The surgery procedures conducted",
+            "type": "string",
+            "uniqueItems": true,
+            "items": {
```
Comment (vsmalladi) [inline src/encoded/schemas/surgery.json]: Move enum up remove items

**Review by vsmalladi — CHANGES_REQUESTED** [review]
Also flip flip y-axis so zero is top left for both medication.

Change facet names for Nephrectomy

**Comment by vsmalladi** [comment]
Closes #51, #9, #96

File: src/encoded/tests/data/inserts/surgery.json (hunk 2 above)
Comment (SenseCode) [reply src/encoded/tests/data/inserts/surgery.json]: Surgery json format change, then branch broke, cannot indexing.

File: src/encoded/schemas/surgery.json (hunk 3 above)
Comment (SenseCode) [reply src/encoded/schemas/surgery.json]: branch broke when json format not the same.

Comment (vsmalladi) [reply src/encoded/schemas/surgery.json]: works now

File: src/encoded/static/components/patient.js [hunk 5]
```diff
@@ -15,149 +15,162 @@ import Radiation from "./radiation";
...
-                <button onClick={this.topFunction} id="scrollUpButton" title="Go to top"><FontAwesomeIcon icon={faAngleDoubleUp} size="2x"/></button>
-            </div>
-        );
+    if (Object.keys(this.props.context.surgery).length > 0) {
+      hasSurgery = true;
     }
+
+    const labsPanelBody = (
+      <PatientChart chartId="labsChart" data={context.labs} ></PatientChart>
+      // <PatientChart chartId="labsChart" data={context.labs} chartTitle ="Lab Results Over Time"></PatientChart>
```
Comment (vsmalladi) [inline src/encoded/static/components/patient.js]: Remove comment

**Review by vsmalladi — APPROVED** [review]

**Pull request merged**